webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/batch/PurgeOldInventoryFiles.scala

```diff
@@ -0,0 +1,121 @@
+ * the GNU General Public License version 3, the copyright holders add
+ * the following Additional permissions:
+ * Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+ * Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+ * Public License version 3, when you create a Related Module, this
+ * Related Module is not considered as a part of the work and may be
+ * distributed under the license agreement of your choice.
+ * A "Related Module" means a set of sources files including their
+ * documentation that, without modification of the Source Code, enables
+ * supplementary functions or services in addition to those offered by
+ * the Software.
+ *
+ * Rudder is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+ *
+ *************************************************************************************
+ */
+
+package com.normation.rudder.batch
+
+import better.files.File
+import com.normation.errors.IOResult
+import com.normation.inventory.domain.InventoryProcessingLogger
+import com.normation.utils.CronParser._
+import com.normation.zio._
+import cron4s.CronExpr
+import java.time.Instant
+import zio._
+import zio.clock.Clock
+import zio.duration._
+
+/**
+ * A scheduler which delete old inventory files under /var/rudder/inventories/{received, failed}
+ * It use a cron-like config parsed by cron4s.
+ */
+class PurgeOldInventoryFiles(
+    optCron:          Option[CronExpr],
+    maxAge:           Duration,
+    cleanDirectories: List[File]
+) {
+
+  val logger = InventoryProcessingLogger
+
+  // check is a file is older than max age (true) or not. Only true for regular file.
+  // maxAge in seconds.
+  def isOlderThanMaxAge(f: File, maxAge: Long, now: Instant): IOResult[Boolean] = {
+    IOResult.effect(f.isRegularFile && f.attributes().creationTime().toInstant.isBefore(now.minusSeconds(maxAge)))
+  }
+
+  // must not fail, will be in a cron
+  val cleanOldFiles: UIO[Unit] = for {
+    t0 <- currentTimeMillis
+    now = Instant.ofEpochMilli(t0)
+    _  <- ZIO
+            .foreach_(cleanDirectories) { dir =>
+              ZIO.foreach_(dir.list.to(Iterable)) { f =>
+                isOlderThanMaxAge(f, maxAge.toSeconds, now).flatMap { older =>
+                  if (older) {
+                    InventoryProcessingLogger.info(
+                      s"Deleting inventory file '${f.pathAsString}' (older than ${maxAge.toString})"
+                    ) *>
+                    IOResult
+                      .effect(f.delete())
+                      .catchAll(err =>
+                        InventoryProcessingLogger.error(s"Error when trying to clean old inventory file '${f.pathAsString}'")
```

Comment:
don't you want to log the stacktrace, at least in debug level?

Reply:
Our users are not developpers. As a rule of thumb, that does not make sens for them, a stacktrace is a bug in the soft. 
So the general approach is to try to make the logs human(ops)-readable.

In that case, I forgot the the `err.fullMsg` though :)